onnxruntime/core/providers/cuda/cuda_graph.cc

```diff
@@ -62,8 +121,22 @@ void CUDAGraph::Reset() {
   }
 }
 
+void CUDAGraph::ResetAdditional() {
```

Comment:
Can this just be part of `Reset()` ?

Reply:
Reset() is used here https://github.com/microsoft/onnxruntime/pull/19636/files/772471c9ca173bdcde3fff99cfb4d6f6d8d73cb8#diff-57ba769b54dce57acd89df47140ede5f29ea670d61176096076701912d573285R208 for session level graph reset()

Reply:
Good point. Not sure if we need to have a `Reset()` in `CaptureBegin()` in the first place. I don't see a reason. Can you see a reason ? If we can just keep `Reset()` (or `Cleanup()`) to be invoked when `CUDAGraph` instance destructs, it will reduce one API (`ResetAdditional()`) - What do you think about this ?

Reply:
sounds good. (we never overwrite a cuda graph in the session level).